fenix/app/src/main/java/org/mozilla/fenix/compose/SwitchWithLabel.kt

```diff
@@ -0,0 +1,179 @@
+        onCheckedChange = onCheckedChange,
+        modifier = modifier,
+        enabled = enabled,
+        colors = SwitchDefaults.colors(
+            uncheckedThumbColor = FirefoxTheme.colors.formOff,
+            uncheckedTrackColor = FirefoxTheme.colors.formSurface,
+            checkedThumbColor = FirefoxTheme.colors.formOn,
+            checkedTrackColor = FirefoxTheme.colors.formSurface,
+            disabledUncheckedThumbColor = FirefoxTheme.colors.formOff
+                .copy(alpha = DISABLED_ALPHA)
+                .compositeOver(FirefoxTheme.colors.formSurface),
+            disabledUncheckedTrackColor = FirefoxTheme.colors.formSurface.copy(alpha = DISABLED_ALPHA),
+            disabledCheckedThumbColor = FirefoxTheme.colors.formOn
+                .copy(alpha = DISABLED_ALPHA)
+                .compositeOver(FirefoxTheme.colors.formSurface),
+            disabledCheckedTrackColor = FirefoxTheme.colors.formSurface.copy(alpha = DISABLED_ALPHA),
+        ),
+    )
+}
+
+@LightDarkPreview
+@Composable
+private fun SwitchWithLabelPreview() {
+    FirefoxTheme {
+        Column(
+            modifier = Modifier
+                .background(FirefoxTheme.colors.layer1)
+                .padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+        ) {
+            Text(
+                text = "Enabled",
+                style = FirefoxTheme.typography.headline7,
+                color = FirefoxTheme.colors.textPrimary,
+            )
+
+            Spacer(Modifier.height(8.dp))
+
+            SwitchWithLabel(
+                checked = false,
+                onCheckedChange = {},
+                label = "Off",
+            )
+
+            SwitchWithLabel(
+                checked = true,
+                onCheckedChange = {},
+                label = "On",
+            )
+
+            Text(
+                text = "Disabled",
+                style = FirefoxTheme.typography.headline7,
+                color = FirefoxTheme.colors.textPrimary,
+            )
+
+            Spacer(Modifier.height(8.dp))
+
+            SwitchWithLabel(
+                checked = false,
+                enabled = false,
+                onCheckedChange = {},
+                label = "Off",
+            )
+
+            SwitchWithLabel(
+                checked = true,
+                enabled = false,
+                onCheckedChange = {},
+                label = "On",
+            )
+
+            Text(
+                text = "Nested",
+                style = FirefoxTheme.typography.headline7,
+                color = FirefoxTheme.colors.textPrimary,
+            )
+
+            Spacer(Modifier.height(8.dp))
+
+            Row {
+                Spacer(Modifier.weight(1f))
+
+                SwitchWithLabel(
+                    checked = false,
+                    onCheckedChange = {},
+                    label = "Nested",
+                    modifier = Modifier.weight(1f),
+                )
+            }
```

Comment:
it might make more sense to have an "isNested" boolean in the API for `SwitchWithLabel`, that way the spacing is consistent across consumers of it. I double checked the figma and our existing nested preferences, and the nested spacing should be 72.

Reply:
hmm, the way I understood the nested component is that there can be parents (Rows) which have their own content on the start (e.g. desktop site menu item) and then `SwitchWithLabel`, and not just a 72.dp padding. 

That seems very much like Settings Screen component and not a generic one, wdyt? The way I think it should be is to be wrapped in another Parent. Also reducing if logic from our components.

Reply:
Hrmm, yeah, I can't immediately think of another example of nested / indented switches besides Settings, so I'm cool with deferring making it part of the API until we have a better understanding of the use cases of this component.